snuba/datasets/dataset.py

```diff
@@ -42,41 +34,34 @@ class Dataset(object):
     manipulate the lower layer objects.
     """
 
-    # TODO: Still not sure if the Dataset class really needs the actual Entity object
-    # or just the name.
     def __init__(self, *, default_entity: Entity) -> None:
-        # TODO: This is a convenience while we slowly migrate everything to Entities. This way
-        # every dataset can have a default entity which acts as a passthrough until we can
-        # migrate the datasets to proper entities.
+        # TODO: Right now there are no Datasets with more than one Entity. Once that changes,
+        # the entity accessor methods will need to be rewritten.
         self.__default_entity = default_entity
 
-    def get_entity(self, entity_name: Optional[str]) -> Entity:
+    def get_entity(self, entity_name: str) -> Entity:
         return self.__default_entity
 
+    def get_default_entity(self) -> Entity:
+        return self.__default_entity
+
+    def get_all_entities(self) -> Sequence[Entity]:
+        return [self.__default_entity]
+
     def get_query_plan_builder(
         self, entity_name: Optional[str] = ""
     ) -> ClickhouseQueryPlanBuilder:
         """
         Returns the component that transforms a Snuba query in a Storage query by selecting
-        the storage and provides the directions on how to run the query.
+        the storage(s) and provides the directions on how to run the query.
         """
-        # TODO: If the query is being executed on a single entity, that entity will be used to determine
-        # the query plan. In cases such as a join, the dataset will something something and
-        # then build the join query.
-        entity = self.get_entity(entity_name)
+        entity = self.get_default_entity()
```

Comment:
Why returning the plan for the default entity no matter on what is passed as argument.

Reply:
Because we have no alternatives right now. This will change when JOINs etc are added.